### PR TITLE
Connection methods

### PR DESCRIPTION
**Pull request by alfred2g**
Add connection methods code snippets for websocket custom auth, proxy, static credentials, username / password

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samples/utils/CommandLineUtils.cpp [hunk 1]
```diff
@@ -677,6 +691,9 @@ namespace Utils
         {
             returnData.input_port = atoi(cmdUtils.GetCommandOrDefault(m_cmd_port_override, "0").c_str());
         }
+        returnData.input_accessKeyId = cmdUtils.GetCommandOrDefault(m_cmd_access_key_id, "");
+        returnData.input_secretAccessKey = cmdUtils.GetCommandOrDefault(m_cmd_secret_access_key, "");
+        returnData.input_sessionToken = cmdUtils.GetCommandOrDefault(m_cmd_session_token, "");
```
Comment (sfod) [inline samples/utils/CommandLineUtils.cpp:696]: Debatable: These options should be required when connecting with static creds. For that, a designated function is needed, which might be overkill. That's why debatable.

**Review by sfod — CHANGES_REQUESTED** [review]

File: samples/mqtt/websocket_connect/README_username_password.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+For this sample, using Websockets will attempt to fetch the AWS credentials to authorize the connection from your environment variables or local files. See the [authorizing direct AWS](https://docs.aws.amazon.com/iot/latest/developerguide/authorizing-direct-aws.html) page for documentation on how to get the AWS credentials, which then you can set to the `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and `AWS_SESSION_TOKEN` environment variables.
+
+</details>
+
+<details>
+<summary> (code snipet to replace similar section)</summary>
+<pre>
+```
+Utils::cmdData cmdData = Utils::parseSampleInputCustomAuthorizerConnect(argc, argv, &apiHandle);
```
[the remark this answers is not in the document]
Comment (alfred2g) [reply samples/mqtt/websocket_connect/README_username_password.md]: The same code is present in the builder code

File: samples/mqtt/websocket_connect/README_proxy.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
+# Websocket Connect with Proxy
```
[the remark this answers is not in the document]
Comment (alfred2g) [reply samples/mqtt/websocket_connect/README_proxy.md:1]: there are 2 functions, one that uses proxy and one only username and password

File: samples/mqtt/websocket_connect/README_static_credentials.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+```
+--access_key_id <str>
+--secret_access_key <str>
+--session_token <str>
+```
+
+To run the websocket connect use the following command:
+
+``` sh
+./websocket-connect --endpoint <endpoint> --signing_region <signing region>
```
[the remark this answers is not in the document]
Comment (alfred2g) [reply samples/mqtt/websocket_connect/README_static_credentials.md:91]: what do you mean? move the above section down?

[the remark this answers is not in the document]
Comment (sfod) [reply samples/mqtt/websocket_connect/README_static_credentials.md:91]: No, I mean the example should contain these options:
```
./websocket-connect --endpoint <endpoint> --signing_region <signing region> --access_key_id <..> ...
```

File: samples/mqtt/websocket_connect/README_proxy.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+</details>
+
+<details>
+<summary> (code snipet to replace the connection_setup function)</summary>
+<pre language="c++"> <code>
+void connection_setup(int argc, char *argv[], ApiHandle &apiHandle, Utils::cmdData &cmdData,
+             Aws::Iot::MqttClientConnectionConfigBuilder &clientConfigBuilder)
+{
+    Utils::cmdData cmdData = Utils::parseSampleInputWebsocketConnect(argc, argv, &apiHandle);
```
Comment (sfod) [inline samples/mqtt/websocket_connect/README_proxy.md:55]: `cmdData` is redefined here, which is a compile-time error. We need to ensure that a code snippet is valid before suggesting it to users

File: samples/mqtt/websocket_connect/README_proxy.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (sfod) [reply samples/mqtt/websocket_connect/README_proxy.md:1]: What I meant is that the only difference between the code snippet provided in this readme and the source file is the proxy option is required in one and optional in another.
Maybe instead just describe this option in the main websocket connection readme?

File: samples/mqtt/websocket_connect/README_proxy.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
+# Websocket Connect with Proxy
+
+[**Return to main sample list**](../../README.md)
+
+This sample makes an MQTT connection via Websockets and then disconnects.
+On startup, the device connects to the server via Websockets and then disconnects right after.
+This sample is for reference on connecting via Websockets using a proxy.
+This sample demonstrates conneting via Websockets using a proxy serverb.
```
Comment (sfod) [inline samples/mqtt/websocket_connect/README_proxy.md:8]: typo: a proxy serverb

File: samples/mqtt/websocket_connect/README_static_credentials.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
+# Websocket Connect with static credentials
+
+[**Return to main sample list**](../../README.md)
+
+This sample makes an MQTT connection via Websockets and then disconnects. On startup, the device connects to the server via Websockets and then disconnects right after. This sample is for reference on connecting via Websockets. This sample demonstrates the most straightforward way to connect via Websockets by querying the AWS credentials for the connection from the device's environment variables or local files.
```
Comment (sfod) [inline samples/mqtt/websocket_connect/README_static_credentials.md:5]: The paragraph should be removed

File: samples/mqtt/websocket_connect/README_username_password.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+--custom_auth_username <str>
+--custom_auth_authorizer_name <str>
+--custom_auth_authorizer_signature <str>
+--custom_auth_password <str>
+```
+
+To run the websocket connect use the following command:
+
+``` sh
+./websocket-connect --endpoint <endpoint> --signing_region <signing region>
```
Comment (sfod) [inline samples/mqtt/websocket_connect/README_username_password.md:84]: `custom_auth_*` option should be added here

File: samples/mqtt/websocket_connect/README_proxy.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (alfred2g) [reply samples/mqtt/websocket_connect/README_proxy.md]: removed it

File: samples/mqtt/websocket_connect/README_custom_auth.md [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+This sample demonstrates connecting via custom auth authorizer username, 
+password and tokens
+
+If you want to use simple or custom auth (or static creds, or basic auth, etc) instead,
+then you will need to replace part of the sample (connection\_setup function) with a code snippet we provided in its corresponding readme.
+
+* [Websocket Connect/simple_auth](./README.md)
+* [Websocket Connect/username and password](./README_username_password.md)
+* [Websocket Connect/proxy](./README_proxy.md)
+* [Websocket Connect/static credentials](./README_static_credentials.md)
```
[the remark this answers is not in the document]
Comment (alfred2g) [reply samples/mqtt/websocket_connect/README_custom_auth.md]: removed

File: samples/mqtt/websocket_connect/README_custom_auth.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+[**Return to main sample list**](../../README.md)
+
+This sample makes an MQTT connection and connects through a Custom Authorizer.
+On startup, the device connects to the server and then disconnects.
+This sample is for reference on connecting using a Custom Authorizer.
+Using a Custom Authorizer allows you to perform your own authorization using an AWS Lambda function.
+See Custom Authorizer for more information.
+You will need to setup your Custom Authorizer so that the lambda function returns a policy document.
+See this page on the documentation for more details and example return result.
```
Comment (sfod) [inline samples/mqtt/websocket_connect/README_custom_auth.md:11]: `this page on the documentation` should be link, like in this readme: https://github.com/aws/aws-iot-device-sdk-java-v2/tree/main/samples/CustomAuthorizerConnect

**Review by sfod — APPROVED** [review]

**Pull request merged**